adobe/aio-app-scripts
# ACNA-464 - full test coverage

## Description
This PR adds **100%** test coverage

It adds the following bug fixes and small changes:
- do not deploy to adobeio default cdn hostname if the user provided his own s3 credentials, but still give the possibility to the user to add his hostname for his configured cdn
- fix get UI url if there is no hostname
- modularize getUIUrl
- fix s3 empty folder for the s3 list truncated case
- modularize fetching main name in package.json
- fix error handling rejection in zip
- add a has JAVA cli check
- wrap connection error on download OW jar
- reduce initial wait for local ow readiness from 4s to 2s
- fix checkFile util
- fix removeProtocolFromUrl function name typo
- build actions pre-build cleanup checks
- build actions with webpack fix error logs when there is also a warning
- fix cleanup of non existing dotenv and .vscode/launch.json file
- fix termination on ctrl+c when no ui server nor local openwhisk server
- fix debugging of UI source maps



## Types of changes




## Checklist:
